README.md

```diff
@@ -1,91 +1,95 @@
 
-This repository contains the smart contracts for validating zero knowledge proofs 
-generated in placeholder proof system in EVM. 
+This repository contains the smart contracts for validating zero-knowledge proofs generated
+via [Placeholder proof system](https://github.com/NilFoundation/evm-placeholder-verification) in-EVM.
 
 ## Dependencies
 
 - [Hardhat](https://hardhat.org/)
-- [nodejs](https://nodejs.org/en/) >= 16.0
+- [nodejs](https://nodejs.org/en/) - Hardhat requires an LTS version; as of August 2023 it's v18.17.0
 
-
-## Clone
-```
+## Clone the project
+```bash
 git clone git@github.com:NilFoundation/evm-placeholder-verification.git
 cd evm-placeholder-verification
 ```
 
 ## Install dependency packages
-```
+```bash
 npm i
 ```
 
 ## Compile contracts
-```
+```bash
 npx hardhat compile
 ```
 
 ## Deploy
 
-Launch a local-network using the following
-```
+Launch a local network using the following command:
+```bash
 npx hardhat node
 ```
 
-To deploy to test environment (ex: Ganache)
-```
-npx hardhat deploy  --network localhost 
+To deploy to a test environment (Ganache, for example), run:
+```bash
+npx hardhat deploy --network localhost
 ```
 
-Hardhat re-uses old deployments, to force re-deploy add the `--reset` flag above
+Hardhat reuses old deployments by default; to force re-deploy, add the `--reset` flag to the command.
 
-## Verification of zkLLVM circtuit compiler output
+## Verification of zkLLVM circuit compiler output
 
-zkLLVM compiler prepares circuits in form of instantiated contracts, which have to be deployed to the blockchain.
+[zkLLVM compiler](https://github.com/NilFoundation/zkllvm) prepares circuits
+as instantiated contracts that can be deployed to a blockchain.
 
-Once you got zkLLVM output - put a circuit directory it under `contracts/zkllvm`. The folder should contain following files:
+Once you get zkLLVM output, place a circuit directory under `contracts/zkllvm`.
+That folder should contain the following files:
 
 ```
-* proof.bin -- placeholder proof file
-* circuit_params.json -- parameters file in JSON format
-* public_input.json -- public input file in JSON format
-* linked_libs_list.json -- list of external libraries, have to be deployed for gate argument computation.
-* gate_argument.sol, gate0.sol, ... gateN.sol -- solidity files with gate argument computation
+* proof.bin — Placeholder proof file
+* circuit_params.json — parameters file
+* public_input.json — public input file
+* linked_libs_list.json — list of external libraries that have to be deployed for gate argument computation
+* gate_argument.sol, gate0.sol, ... gateN.sol — Solidity files with gate argument computation
 ```
 
-If all the files are in place - you can deploy the circuit verifier to the blockchain and verify the proofs.
+If all these files are in place, you can deploy the circuit verifier to the blockchain and verify the proofs.
```

Comment:
Do I need a proof file to deploy the circuit verifier? Do I need to deploy the verifier? What's a circuit verifier?

Reply:
Soo
To deploy the Verifier contract locally or onto the blockchain you need:
* linked_libs_list.json — list of external libraries that have to be deployed for gate argument computation
* gate_argument.sol, gate0.sol, ... gateN.sol — Solidity files with gate argument computation

To use the Verifier contract, you need:
* proof.bin — Placeholder proof file
* circuit_params.json — parameters file
* public_input.json — file with public input
* the address to the deployed gate argument(s)